File: example/src/main/java/ru/virgil/example/box/Box.java
package ru.virgil.example.box;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import ru.virgil.example.system.IdentifiedEntity;
import ru.virgil.example.truck.Truck;
import ru.virgil.example.user.UserDetails;

import javax.persistence.*;

@Entity
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
public class Box extends IdentifiedEntity {

    @ManyToOne
    private UserDetails owner;
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private BoxType type;
    @ManyToOne(cascade = CascadeType.REMOVE)
    private Truck truck;
    private String description;
    private int price;
    private float weight;

}

File: example/src/main/java/ru/virgil/example/box/Box.kt
package ru.virgil.example.box

import ru.virgil.example.system.entity.OwnedEntity
import ru.virgil.example.truck.Truck
import ru.virgil.example.user.UserDetails
import javax.persistence.*

@Entity
class Box(
    owner: UserDetails,
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var type: BoxType = BoxType.USUAL,
    @ManyToOne(cascade = [CascadeType.REMOVE])
    var truck: Truck,
    var description: String,
    var price: Int = 0,
    var weight: Float = 0f,
) : OwnedEntity(owner)
